internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,604 @@
+			SuccessThreshold:    1,
+			InitialDelaySeconds: 3,
+			PeriodSeconds:       10,
+			TimeoutSeconds:      3,
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{
+					Port: intstr.FromInt(int(servicePort)),
+				},
+			},
+		},
+	}
+
+	postgresContainer := corev1.Container{
+		Name:            databaseContainerName,
+		Image:           DatabaseImage(),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          databaseName,
+				ContainerPort: databasePort,
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: []corev1.EnvVar{
+			newSecretEnvVar("POSTGRESQL_DATABASE", "db.name", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_USER", "db.user", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_PASSWORD", "db.password", databaseSecretName),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "postgresdb",
+				MountPath: "/var/lib/pgsql/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(200, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(500*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
+			},
+		},
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: "bucket-filesystem",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: filesystemPVCName,
+				},
+			},
+		},
+		{
+			Name: "postgresdb",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: databasePVCName,
+				},
+			},
+		},
+	}
+
+	deploymentLabels := map[string]string{
+		"app": assistedServiceDeploymentName,
+	}
+
+	deploymentStrategy := appsv1.DeploymentStrategy{
+		Type: appsv1.RollingUpdateDeploymentStrategyType,
+		RollingUpdate: &appsv1.RollingUpdateDeployment{
+			MaxUnavailable: &maxUnavailable,
+			MaxSurge:       &maxSurge,
+		},
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      assistedServiceDeploymentName,
+			Namespace: r.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: deploymentLabels,
+					Name:   assistedServiceDeploymentName,
+				},
+			},
+		},
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, deployment, r.Scheme); err != nil {
+			return err
+		}
+		var replicas int32 = 1
+		deployment.Spec.Replicas = &replicas
+		deployment.Spec.Strategy = deploymentStrategy
+		deployment.Spec.Template.Spec.Containers = []corev1.Container{serviceContainer, postgresContainer}
```

Comment:
Why create a service for PG if we're running in the same pod? Couldn't we use some sort of local communication at that point?

Reply:
Going to resolve this as we're using `localhost`.
If we continue to run the assisted-service and postgres in the same pod we might want to look into using a local socket and closing the port on the PG container, but that's out of scope for this first patch and would probably require editing some additional PG config.

Reply:
Postgres service has been removed.